core/types/timed_cache.go

```diff
@@ -0,0 +1,177 @@
+// Package types contains data types related to Ethereum consensus.
+package types
+
+import (
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// timedEntry provides a wrapper to store an entry in an LRU cache, with a
+// specified expiration time
+type timedEntry struct {
+	expiresAt int64
+	value     interface{}
+}
+
+// expired returns whether or not the given entry has expired
+func (te *timedEntry) expired() bool {
```

Comment:
Would it not be more efficient to create a tc interface around the lru object?

Reply:
Problem is it needs extra data (a ttl field) which does not exist in the LRU object. I saw two options:
1) fork lru and add ttl behavior
2) add a new cache type which extended the functionality of the lru cache

I opted for no. 2, obviously, but we could consider no. 1 if you prefer.

Reply:
If this is what we think the efficient choice is that is fine.  I just hate seeing code duplication.

Reply:
Well the point of wrapping was to avoid duplication. What is duplicated?

Reply:
NVM.  This is fine.